src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BindLoweringPass.cs

```diff
@@ -0,0 +1,419 @@
+            changeHandlerAttributeName = bindTagHelper.GetChangeHandlerAttributeName() ?? changeHandlerAttributeName;
+
+            // We expect 0-1 components per-node.
+            var componentTagHelper = node.TagHelpers.FirstOrDefault(t => t.IsComponentTagHelper());
+            if (componentTagHelper == null)
+            {
+                // If it's not a component node then there isn't too much else to figure out.
+                return attributeName != null && changeHandlerAttributeName != null;
+            }
+
+            // If this is a component, we need an attribute name for the value.
+            if (attributeName == null)
+            {
+                return false;
+            }
+
+            // If this is a component, then we can infer '<PropertyName>Changed' as the name
+            // of the change handler.
+            if (changeHandlerAttributeName == null)
+            {
+                changeHandlerAttributeName = valueAttributeName + "Changed";
+            }
+
+            for (var i = 0; i < componentTagHelper.BoundAttributes.Count; i++)
+            {
+                var attribute = componentTagHelper.BoundAttributes[i];
+
+                if (string.Equals(valueAttributeName, attribute.Name))
+                {
+                    valueAttribute = attribute;
+                }
+
+                if (string.Equals(changeHandlerAttributeName, attribute.Name))
+                {
+                    changeHandlerAttribute = attribute;
+                }
+            }
+
+            return true;
+        }
+
+        private bool TryGetFormatNode(
+            TagHelperIntermediateNode node,
+            ComponentAttributeExtensionNode attributeNode,
+            string valueAttributeName,
+            out ComponentAttributeExtensionNode formatNode)
+        {
+            for (var i = 0; i < node.Children.Count; i++)
+            {
+                var child = node.Children[i] as ComponentAttributeExtensionNode;
+                if (child != null &&
+                    child.TagHelper != null &&
+                    child.TagHelper == attributeNode.TagHelper &&
+                    ((child.AttributeName == "format" && attributeNode.AttributeName == "bind") ||
+                    (child.AttributeName == "format-" + valueAttributeName)))
+                {
+                    formatNode = child;
+                    return true;
+                }
+            }
+
+            formatNode = null;
+            return false;
+        }
+
+        private static string GetAttributeContent(ComponentAttributeExtensionNode node)
+        {
+            if (node.Children[0] is HtmlContentIntermediateNode htmlContentNode)
+            {
+                // This case can be hit for a 'string' attribute. We want to turn it into
+                // an expression.
+                return "\"" + ((IntermediateToken)htmlContentNode.Children.Single()).Content + "\"";
```

Comment:
Oh wait, I see from the tests below that `bind="MyProperty"` *is* supported. Nice :)

Reply:
Yeah, regarding this I think we need a language feature that allows tag helpers to specify whether they want the attribute's context to be HTML or C#. Right now this is controlled by the type, so if it's a string attribute, the `@` is always required to use a variable/expression because you're in an HTML context by default. I think we want  authors to be able to control this.